### PR TITLE
Only codegen classes that are actually used

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [X] ~~I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?
* Filter converters and attribute classes based on what the driver actually uses
    * Should improve code coverage numbers
    * Didn't try to do this for matchers - that would be more difficult to calculate which ones are used
    * Required moving (and possibly duplicating) imports into class/function that actually uses it
* Only monitor the files we care about from nimodinst
### What testing has been done?
* Unit
* System

### CODE REVIEW

**Comment by marcoskirsch** [comment]
PR description:

> Should improve code coverage numbers

Codecov report:

> Merging #1039 into master will decrease coverage by 0.56%.

What gives?

File: build/helper/metadata_add_all.py [hunk 1]
```diff
@@ -321,7 +321,7 @@ def _setup_init_function(functions, config):
         # Change the init_function information for generating the docstring
         # We are assuming the last parameter is vi out
         for p in init_function['parameters']:
-            if p['name'] == config['session_handle_parameter_name']:
+            if p['name'] == 'vi':
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/metadata_add_all.py]: I don't know. This branch is up to date from master.

Fixed.

File: build/helper/metadata_add_all.py [hunk 2]
```diff
@@ -339,6 +339,15 @@ def _setup_init_function(functions, config):
         pass
 
 
+def _add_param_usages(p, config):
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/metadata_add_all.py]: Originally I was going to add which matchers were used as well, but that wasn't straight forward, so for parameters it ended up only converter names used.

Changed name to `_add_param_converters_used` and fixed comment.

File: build/helper/metadata_add_all.py [hunk 3]
```diff
@@ -411,6 +421,31 @@ def _add_repeated_capability_type(a, attributes):
         attributes[a]['repeated_capability_type'] = 'channels'
 
 
+def _add_attribute_usages(a, config):
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/metadata_add_all.py]: Changed name to `_add_attribute_types_converters_used`

File: .codecov.yml [hunk 4]
```diff
@@ -6,10 +6,20 @@ coverage:
...
       default:
-        target: 90%
+        target: 85%
         threshhold: 5
     patch:
       default:
         target: 70%
         threshhold: 5
 
+ignore:
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply .codecov.yml]: Added to PR description

**Comment by texasaggie97** [comment]
> This eliminates a bunch of dead code... but is it worth the added complexity?

That is a good question. Honestly, I thought it would make a bigger difference in code coverage than it is. I would say it makes the code more correct.

Given the (lack of) improvement in code coverage, I am fine with abandoning this change and just leave it as is with lots of unused code in each driver, even though it would be more correct.

**Comment by texasaggie97** [comment]
Abandoning since it is not worth the extra complexity.

**Pull request closed without merging**